### DISCUSSION TITLE
🍵 Matcha - A new Catppuccin flavor

### DISCUSSION THREAD

**Discussion by UsamaKhan** [catppuccin/catppuccin#2584] [Polls]
# 🍵 Matcha - A new Catppuccin flavor

Hi everyone! I'd like to propose a new flavor for Catppuccin called "Matcha". It's designed to be a dark theme with a green undertone, complementing the existing flavors while maintaining the Catppuccin design philosophy.

## 🧠 Design Philosophy Alignment

This flavor follows Catppuccin's core principles (from the [README](https://github.com/catppuccin/catppuccin/blob/main/README.md#-design-philosophy)):
- **Colorful yet balanced**: Uses vibrant accents while maintaining readability
- **Harmonious**: Green undertones create a cohesive look
- **Middle-ground contrast**: Balances readability with visual comfort

## 🎨 Color Palette

The palette consists of the standard 26-color structure with a green-tinted monochromatic base:

### Accent Colors

| Label     | Hex       | RGB                  | HSL                  |
| --------- | --------- | -------------------- | -------------------- |
| Rosewater | `#E3B197` | `rgb(227, 177, 151)` | `hsl(22, 60%, 74%)`  |
| Flamingo  | `#E3A097` | `rgb(227, 160, 151)` | `hsl(7, 60%, 74%)`   |
| Pink      | `#F0A8D1` | `rgb(240, 168, 209)` | `hsl(324, 73%, 80%)` |
| Mauve     | `#B58FEA` | `rgb(181, 143, 234)` | `hsl(262, 72%, 74%)` |
| Red       | `#E67D7D` | `rgb(230, 125, 125)` | `hsl(0, 69%, 70%)`   |
| Maroon    | `#E68F8F` | `rgb(230, 143, 143)` | `hsl(0, 65%, 73%)`   |
| Peach     | `#E6A367` | `rgb(230, 163, 103)` | `hsl(28, 72%, 65%)`  |
| Yellow    | `#DCC26A` | `rgb(220, 194, 106)` | `hsl(49, 62%, 64%)`  |
| Green     | `#9ECB8F` | `rgb(158, 203, 143)` | `hsl(105, 37%, 68%)` |
| Teal      | `#8BCBB5` | `rgb(139, 203, 181)` | `hsl(162, 39%, 67%)` |
| Sky       | `#89CFE5` | `rgb(137, 207, 229)` | `hsl(195, 65%, 72%)` |
| Sapphire  | `#75B5CC` | `rgb(117, 181, 204)` | `hsl(197, 45%, 63%)` |
| Blue      | `#86A9E9` | `rgb(134, 169, 233)` | `hsl(221, 70%, 72%)` |
| Lavender  | `#B5BEF5` | `rgb(181, 190, 245)` | `hsl(231, 77%, 84%)` |

### Monochrome Colors

| Label    | Hex       | RGB                  | HSL                  |
| -------- | --------- | -------------------- | -------------------- |
| Text     | `#D5E0DC` | `rgb(213, 224, 220)` | `hsl(153, 16%, 86%)` |
| Subtext1 | `#BCC5C1` | `rgb(188, 197, 193)` | `hsl(153, 10%, 75%)` |
| Subtext0 | `#A3ABA7` | `rgb(163, 171, 167)` | `hsl(153, 7%, 65%)`  |
| Overlay2 | `#8A918D` | `rgb(138, 145, 141)` | `hsl(153, 5%, 55%)`  |
| Overlay1 | `#717873` | `rgb(113, 120, 115)` | `hsl(153, 3%, 45%)`  |
| Overlay0 | `#585F5A` | `rgb(88, 95, 90)`    | `hsl(153, 4%, 36%)`  |
| Surface2 | `#3F4641` | `rgb(63, 70, 65)`    | `hsl(153, 5%, 26%)`  |
| Surface1 | `#363D38` | `rgb(54, 61, 56)`    | `hsl(153, 6%, 23%)`  |
| Surface0 | `#2D342F` | `rgb(45, 52, 47)`    | `hsl(153, 7%, 19%)`  |
| Base     | `#242B26` | `rgb(36, 43, 38)`    | `hsl(153, 9%, 15%)`  |
| Mantle   | `#1B221D` | `rgb(27, 34, 29)`    | `hsl(153, 11%, 12%)` |
| Crust    | `#121914` | `rgb(18, 25, 20)`    | `hsl(153, 16%, 8%)`  |

### ANSI Colors for Matcha

| ANSI Color     | Hex       | RGB                  | HSL                  |
| -------------- | --------- | -------------------- | -------------------- |
| Bright Black   | `#3F4641` | `rgb(63, 70, 65)`    | `hsl(153, 5%, 26%)`  |
| Bright Red     | `#E67575` | `rgb(230, 117, 117)` | `hsl(2, 69%, 66%)`   |
| Bright Green   | `#96C587` | `rgb(150, 197, 135)` | `hsl(107, 45%, 64%)` |
| Bright Yellow  | `#D4BC62` | `rgb(212, 188, 98)`  | `hsl(51, 70%, 60%)`  |
| Bright Blue    | `#7EA3E3` | `rgb(126, 163, 227)` | `hsl(223, 78%, 68%)` |
| Bright Magenta | `#E8A2CB` | `rgb(232, 162, 203)` | `hsl(326, 81%, 76%)` |
| Bright Cyan    | `#83C5AF` | `rgb(131, 197, 175)` | `hsl(164, 47%, 63%)` |
| Bright White   | `#BCC5C1` | `rgb(188, 197, 193)` | `hsl(153, 10%, 75%)` |

### Semantic Colors (60% Text/Color Mixes)

| Label           | Hex       | RGB                  | HSL                  |
| --------------- | --------- | -------------------- | -------------------- |
| Semantic Red    | `#DDB7B7` | `rgb(221, 183, 183)` | `hsl(0, 35%, 79%)`   |
| Semantic Yellow | `#D9D1B1` | `rgb(217, 209, 177)` | `hsl(49, 31%, 77%)`  |
| Semantic Green  | `#BFD1BA` | `rgb(191, 209, 186)` | `hsl(105, 19%, 77%)` |
| Semantic Teal   | `#B8D1CA` | `rgb(184, 209, 202)` | `hsl(162, 20%, 77%)` |
| Semantic Blue   | `#B8C7DC` | `rgb(184, 199, 220)` | `hsl(221, 35%, 79%)` |
| Semantic Mauve  | `#C8BFE1` | `rgb(200, 191, 225)` | `hsl(262, 36%, 82%)` |

## 📸 Preview

![image](https://github.com/user-attachments/assets/79229e1e-b0b4-45a3-a15f-bd4e804d0658)

## 📝 Implementation Details

The theme follows all Catppuccin conventions:
- Maintains 26-color structure
- Uses HSL-based color relationships
- Follows semantic color usage guidelines
- Implements proper ANSI color mapping

## 💬 Community Feedback

I'd love to hear everyone's thoughts on:
1. Fit within the Catppuccin family
2. Overall color balance
3. Readability and contrast
4. Potential improvements
5. Use cases where this flavor might excel

Looking forward to the feedback! 🍵✨

**Comment by unseen-ninja** [comment]
I saw the suggestion on Discord but couldn't quite imagine how it would look like; turns out it actually looks better than I pictured in my head. Comparing it to the existing flavours I'm not entirely sold that it fits into _that_ lineup, but on it's own – and potentially with the Matcha Latte variant as well – this is a solid derivative theme.

**Comment by sgoudham** [comment]
Hey :wave:,

Thanks for sharing, looks like a nice derivation of Catppuccin but as @unseen-ninja mentioned above, I don't believe it would really fit into the current line-up of flavours.

Not to mention that I, personally, am pushing back on the idea of new flavours (or even big highlight reworks) until the organisation is in a much better state where ports can be updated in a far easier/less tedious fashion as part of "lessons learned" from the huge amount of manual effort it took for v0.1.0 -> v0.2.0. 

I'll be closing this discussion as I'm not sure if there's much more to discuss but thanks again for sharing!
